### PR TITLE
Pass CLAIM_URI_LOCALE value from identity.xml

### PR DESCRIPTION
**Pull request by Ammoniya**
### Proposed changes in this pull request

Update the value of `CLAIM_URI_LOCALE` to  `http://wso2.org/claims/local`, valued passed from identity.xml

Related issue - https://github.com/wso2/product-is/issues/12163

* * *
### Note
:bulb: This PR's builder will be failed till bump the identity framework version

### CODE REVIEW

File: components/org.wso2.carbon.identity.application.authenticator.live/src/main/java/org/wso2/carbon/identity/application/authenticator/live/WindowsLiveOAuth2AuthenticatorConstants.java [hunk 1]
```diff
@@ -33,8 +35,19 @@
     public static final String GIVEN_NAME_CLAIM_URI = "http://wso2.org/claims/givenname";
     public static final String LAST_NAME_CLAIM_URI = "http://wso2.org/claims/lastname";
     public static final String GENDER_CLAIM_URI = "http://wso2.org/claims/gender";
-    public static final String LOCALITY_CLAIM_URI = "http://wso2.org/claims/locality";
+    public static final String LOCALITY_CLAIM_URI = getClaimUriLocale();
 
     private WindowsLiveOAuth2AuthenticatorConstants() {
     }
+
+    private static String getClaimUriLocale() {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.live/src/main/java/org/wso2/carbon/identity/application/authenticator/live/WindowsLiveOAuth2AuthenticatorConstants.java]: Will we be using the same logic in multiple places?

Will there be any issue in introducing a common place to get this logic and reusing it as oppose to implementing it in multiple places?

File: components/org.wso2.carbon.identity.application.authenticator.live/src/main/java/org/wso2/carbon/identity/application/authenticator/live/WindowsLiveOAuth2AuthenticatorConstants.java [hunk 2]
```diff
@@ -33,8 +35,19 @@
...
+
+        if (Boolean.parseBoolean(IdentityUtil.getProperty("UseLegacyLocalizationClaim"))) {
+            return "http://wso2.org/claims/locality";
+        } else if (IdentityUtil.getProperty("UseLegacyLocalizationClaim") == null) {
+            return "http://wso2.org/claims/local";
```
Comment (mevan-karu) [inline components/org.wso2.carbon.identity.application.authenticator.live/src/main/java/org/wso2/carbon/identity/application/authenticator/live/WindowsLiveOAuth2AuthenticatorConstants.java]: `Boolean.parseBoolean(IdentityUtil.getProperty("UseLegacyLocalizationClaim"))` returns false when `IdentityUtil.getProperty("UseLegacyLocalizationClaim") == null`. Isn't this redundant?

File: components/org.wso2.carbon.identity.application.authenticator.live/src/main/java/org/wso2/carbon/identity/application/authenticator/live/WindowsLiveOAuth2AuthenticatorConstants.java (hunk 1 above)
Comment (Ammoniya) [reply components/org.wso2.carbon.identity.application.authenticator.live/src/main/java/org/wso2/carbon/identity/application/authenticator/live/WindowsLiveOAuth2AuthenticatorConstants.java]: Fixed in commit 4734163c36579cd69deab666ff3bed691a1dd8ca

File: components/org.wso2.carbon.identity.application.authenticator.live/src/main/java/org/wso2/carbon/identity/application/authenticator/live/WindowsLiveOAuth2AuthenticatorConstants.java (hunk 2 above)
Comment (Ammoniya) [reply components/org.wso2.carbon.identity.application.authenticator.live/src/main/java/org/wso2/carbon/identity/application/authenticator/live/WindowsLiveOAuth2AuthenticatorConstants.java]: Correct, its redundant
Fixed in - https://github.com/wso2/carbon-identity-framework/pull/3634/commits/1da6face9ef31e4f037c27aff176ef51dd25d85b

**Review by emswbandara — APPROVED** [review]

**Comment by emswbandara** [comment]
update framework version before merging.

**Pull request merged**